modelservice/games/scopes/base.py

```diff
@@ -305,22 +305,29 @@ def _scope_tree(self, *args, **kwargs):
         scope_groups = self.child_scopes
         for resource_name, scope_group in scope_groups.items():
             self.log.debug(
-                '_scope_tree: resource_name: {name}, user: {user!s}',
-                name=resource_name, user=user)
+                '_scope_tree: resource_name: {name}, user: {user!s}, exclude: {exclude}',
+                name=resource_name, user=user, exclude=exclude)
 
-            children = scope_group.for_user(user)
+            if resource_name == exclude:
+                self.log.debug(
+                    '_scope_tree: exclude {name} children', name=resource_name)
+            else:
+                children = scope_group.for_user(user)
 
-            self.log.debug('_scope_tree: children: {children!s}',
-                           children=children)
+                self.log.debug('_scope_tree: children: {children!s}',
+                               children=children)
 
-            payload['children'] += [child._scope_tree(*args, **kwargs) for
-                                    child in children]
+                payload['children'] += \
+                    [child._scope_tree(exclude, *args, **kwargs)
+                     for child in children]
         return payload
 
     @register
     def get_scope_tree(self, *args, **kwargs):
-        self.log.debug('get_scope_tree: {name} pk: {pk}',
-                       name=self.resource_name, pk=self.pk)
+        exclude = args[0] if len(args) > 0 else None
```

Comment:
I'd rather have `exclude` in the signature: `def get_scope_tree(self, exclude=None, *args, **kwargs):`

Reply:
That's what I tried first, but it didn't work with the simpl.getDataTree action. Is there a way to define default values for actions created with createAction?

Reply:
how did you call the action? IIRC the syntax can be a little awkward on the JS side

Reply:
have you tried `getDataTree(topic, [], {exclude: 'result'})`?

Reply:
You could also try these:
`getDataTree(topic, ['result'])`
`getDataTree(topic, {args: [], kwargs: {exclude: 'result'}})`
`getDataTree(topic, {args: ['result']})`

Reply:
Added an explicit exclude parameter to getDataTree to pass an array of resource names to be excluded.